plugins/sqs/sqs.cpp

```diff
@@ -0,0 +1,539 @@
+  Aws::Client::ClientConfiguration* config= new Aws::Client::ClientConfiguration();
+ if(!protocol.empty())
+  {
+    config->scheme = Aws::Http::Scheme::HTTPS;
+  }
+
+  if(region.empty()) throw string("Region mustn't be empty");
+  if(isRegionExist(region))
+    {
+      config->region = getRegion(region);
+    }
+
+  this->sqsClient= new Aws::SQS::SQSClient(*config);
+
+ //this->queueUrl=getRegion(region);
+   cout << "Queue URL is "  << this->queueUrl << endl;
+// this->setQueueUrlFromQueueName();
+ 
+ delete(config);
+}
+
+
+
+/**
+
+
+ **/
+bool SQSHPCCPlugin::SQSHPCC::isRegionExist(const string& region)
+{
+  
+  if(region.empty())
+    {
+      return false;
+    }
+
+  char *reg = new char[region.length() +1];
+  strcpy(reg,region.c_str());
+  upstr(reg);
+ 
+  if((strncmp(reg,"US_EAST_1",9)==0)|| (strncmp(reg,"US_WEST_1",9)==0) || (strncmp(reg,"EU_WEST_1",9)==0)
+      || (strncmp(reg,"EU_CENTRAL_1",12)==0) || (strncmp(reg,"AP_SOUTHEAST_1",14)==0) || (strncmp(reg,"AP_SOUTHEAST_2",14)==0)
+     ) return true;
+  
+  return false;
+  
+}
+
+
+/**
+*
+*
+*/
+const char *const SQSHPCCPlugin::SQSHPCC::getRegion(const string& region) 
+{
+
+  char *reg = new char[region.length() +1];
+  strcpy(reg,region.c_str());
+  upstr(reg);
+  puts(reg);
+    
+  if(strncmp(reg,"US_EAST_1",9)==0) return Aws::Region::US_EAST_1;
+  if(strncmp(reg,"US_WEST_1",9)==0) return Aws::Region::US_WEST_1;
+  if(strncmp(reg,"EU_WEST_1",9)==0) return Aws::Region::EU_WEST_1;
+  if(strncmp(reg,"EU_CENTRAL_1",12)==0) return Aws::Region::EU_CENTRAL_1;
+  if(strncmp(reg,"AP_SOUTHEAST_1",14)==0) return Aws::Region::AP_SOUTHEAST_1;
+  if(strncmp(reg,"AP_SOUTHEAST_2",14)==0)  return Aws::Region::AP_SOUTHEAST_2;
+
+  return NOTFOUND;
+}
+
+
+ void SQSHPCCPlugin::SQSHPCC::upstr(char *s)
+{
+  char  *p;
+
+  for (p = s; *p != '\0'; p++) 
+    *p = (char) toupper(*p);
+}
+
+
+
+string SQSHPCCPlugin::SQSHPCC::convertAwsStringToCharPtr(Aws::String str)
+{
+
+  //char* chr = new char[str.length()+1];
+ // strcpy(chr,str.c_str());
+  string res(str.c_str());
+  return res;
+}
+
+/**
+*
+*
+*/
+
+namespace SQSHPCCPlugin
+{
+//ECL_SQS_API bool ECL_SQS_CALL publishMessage(ICodeContext* ctx, const char* region, const char* queueName, const char* message)
+ECL_SQS_API bool ECL_SQS_CALL publishMessage(ICodeContext * ctx,const char* region, const char* queueName, const char* message)
+{
+
+   if(strlen(queueName) == 0) 
+	{
+	    cout << "QueueName is Empty" << endl;
+            throw runtime_error("The queueName mustn't be empty!!!");
+	}
+   
+    try 
+	{ 
+          SQSHPCCPlugin::SQSHPCC* hpcc= new SQSHPCCPlugin::SQSHPCC(queueName);
+         hpcc->setSQSConfiguration("HTTPS",region);
+     	 hpcc->setQueueUrlFromQueueName();
+         SQSHPCCPlugin::Response response = hpcc->sendMessage(message);
+          
+//         cout << "The result is " << ref.body << endl; 
+         //return (res.code == 3);
+       //  hpcc->~SQSHPCC();
+         delete(hpcc);
```

Comment:
This object is going to leak if there is an exception. Not sure why it needs to be allocated dynamically - a local object of type SQSHPCCPlugin::SQSHPCC would be safer (and more efficient)